File: src/main/java/gargoyle/netsquares/store/prop/PrefIntegerProperty.java
package gargoyle.netsquares.store.prop;

public class PrefIntegerProperty extends BasePrefProperty<Integer> {
    public PrefIntegerProperty(Class<?> pkg, String key, Integer defaultValue) {
        super(pkg, key, defaultValue);
    }

    @Override
    protected void _set(Integer value) {
        preferences.putInt(key, value);
    }

    public Integer get(Integer defaultValue) {
        return preferences.getInt(key, defaultValue);
    }

    public void setValue(int value) {
        set(value);
    }
}

File: src/main/kotlin/gargoyle/netsquares/store/prop/PrefIntegerProperty.kt
package gargoyle.netsquares.store.prop

import kotlin.reflect.KClass

class PrefIntegerProperty(pkg: KClass<*>, key: String, defaultValue: Int) :
    BasePrefProperty<Int>(pkg, key, defaultValue) {
    override fun _set(value: Int) {
        preferences.putInt(key, value)
    }

    override fun get(defaultValue: Int): Int {
        return preferences.getInt(key, defaultValue)
    }

    fun setValue(value: Int) {
        set(value)
    }
}
